### PR TITLE
n and p parametrization on Zero Inflated Negative Binomial

### PR DESCRIPTION
**Pull request by farhanreynaldo**
This PR added alternative parametrization (`n` and `p`) on Zero Inflated Negative Binomial based on this issue https://github.com/pymc-devs/pymc/issues/5196.

### CODE REVIEW

File: pymc/tests/test_distributions_random.py [hunk 1]
```diff
@@ -1502,6 +1502,40 @@ def seeded_zero_inflated_negbinomial_rng_fn(self):
     ]
 
 
+class TestZeroInflatedNegativeBinomial(BaseTestDistribution):
```
Comment (ricardoV94) [inline pymc/tests/test_distributions_random.py:1505]: This test can be much smaller. Check other tests for alternative parametrizations in this module. Like `NormalTau`

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Just did a quick skim. Looks great so far

File: pymc/tests/test_distributions_random.py (hunk 1 above)
Comment (farhanreynaldo) [reply pymc/tests/test_distributions_random.py:1505]: After a quick glance on `NormalTau`, we only need to test on `check_pymc_params_match_rv_op`?

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py:1505]: Yes

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Some of the tests seen to be failing.

**Comment by farhanreynaldo** [comment]
> Some of the tests seen to be failing.

This line is failing:

```
self.check_logcdf(
    ZeroInflatedNegativeBinomial,
    Nat,
    {"psi": Unit, "p": Unit, "n": NatSmall},
    lambda value, psi, p, n: np.log(
        (1 - psi) + psi * sp.nbinom.cdf(value, n, p)
    ),
)
```

Did I misspecify the `{"psi": Unit, "p": Unit, "n": NatSmall}` this one?

**Comment by ricardoV94** [comment]
From the trace it seems the distribution is not respecting the invalid negative `n` parameter: https://github.com/pymc-devs/pymc/runs/4276276491?check_suite_focus=true#step:8:429

**Comment by ricardoV94** [comment]
Which is strange because it passes with the mu/alpha parametrization

**Comment by ricardoV94** [comment]
Seems like we have a small bug in the logcdf. We have to add a `0 < n` bound condition to the logcdf here: https://github.com/pymc-devs/pymc/blob/dc92865f906e20d6ae9970f49671ae0accf9779a/pymc/distributions/discrete.py#L1710

The case `n <= 0` was not being tested independently before apparently. Perhaps because it led to a negative `p` (when converting from `mu`) which was enough to activate the bound conditions.

**Comment by farhanreynaldo** [comment]
> Seems like we have a small bug in the logcdf. We have to add a `0 < n` bound condition to the logcdf here:
> 
> https://github.com/pymc-devs/pymc/blob/dc92865f906e20d6ae9970f49671ae0accf9779a/pymc/distributions/discrete.py#L1710
> 
> The case `n<0` was not being tested independently before apparently. Perhaps because it led to a negative `p` (when converting from `mu`) which was enough to activate the bound conditions.

I am wondering, so in order to test the `n` and `p` parametrization, the `logcdf` bound (i.e. `n<0`) should be written explicitly. Why can't we rely on the implicit case where negative `n` would yield negative `p`, which activate the `0 < p` bounds?

**Comment by ricardoV94** [comment]
> I am wondering, so in order to test the `n` and `p` parametrization, the `logcdf` bound (i.e. `n<0`) should be written explicitly. Why can't we rely on the implicit case where negative `n` would yield negative `p`, which activate the `0 < p` bounds?

We are not computing `p` from `n` because the new parametrization is already given in terms of `p`.

**Comment by ricardoV94** [comment]
Thanks @farhanreynaldo!

**Comment by farhanreynaldo** [comment]
> > I am wondering, so in order to test the `n` and `p` parametrization, the `logcdf` bound (i.e. `n<0`) should be written explicitly. Why can't we rely on the implicit case where negative `n` would yield negative `p`, which activate the `0 < p` bounds?
> 
> We are not computing `p` from `n` because the new parametrization is already given in terms of `p`.

Ah right.

**Pull request merged**